### PR TITLE
Kernel: Remove hardcoded ipython kernel

### LINKED ISSUE #203: Kernel hardcode
Anybody knows is this lines actual?
https://github.com/nteract/hydrogen/blob/master/lib/kernel.coffee#L33-L45

This clobber should be removed.

### PR DESCRIPTION
**Pull request by lgeiger**
Fixes #203

### CODE REVIEW

File: lib/kernel.coffee [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Kernel
         console.log 'Kernel spec:', @kernelSpec
         console.log 'Kernel configuration:', @config
         console.log 'Kernel configuration file path:', @configPath
-        @language = @kernelSpec.grammarLanguage
+        @language = @kernelSpec.language
```
Comment (lgeiger) [inline lib/kernel.coffee]: `@language` was always `undefined`. Hence the hardcoded ipython kernel couldn't ever work.

I think we can remove it because nobody reported an issue 😉 .

Comment (n-riesco) [inline lib/kernel.coffee]: We need to remove [this line](https://github.com/nteract/hydrogen/blob/e3927895dc903d408e5fe1624ff3e81cdaed3b59/lib/signal-list-view.coffee#L63) as well.

**Comment by n-riesco** [comment]
We should provide a sample kernel spec for users like Ubuntu 16.04's that don't have `ipython kernelspec`. For example:

```json
{
  "kernelspecs": {
    "python2": {
      "spec": {
        "display_name": "Python 2",
        "language": "python",
        "argv": ["python2.7", "-m", "ipykernel", "-f", "{connection_file}"],
        "env": {}
      }
    },
    "python3": {
      "spec": {
        "display_name": "Python 3",
        "language": "python",
        "argv": ["python3.4", "-m", "ipykernel", "-f", "{connection_file}"],
        "env": {}
      }
    }
  }
}
```

File: lib/kernel.coffee (hunk 1 above)
Comment (n-riesco) [inline lib/kernel.coffee]: After some more thought and since `Kernel` gets a `grammar`, I think this line should be `@language =  KernelManager::getLanguageFor grammar`.

This would also suggest that  `getLanguageFor` should be a static method (I would leave that for another PR because `getLanguageFor` is used everywhere).

Comment (lgeiger) [inline lib/kernel.coffee]: > We need to remove this line as well.
Done in #339 

> After some more thought and since Kernel gets a grammar, I think this line should be @language = KernelManager::getLanguageFor grammar.
Fixed!

**Comment by lgeiger** [comment]
> We should provide a sample kernel spec for users like Ubuntu 16.04's that don't have `ipython kernelspec`.

Yes this should be added to the readme.

File: lib/kernel.coffee [hunk 2]
```diff
@@ -15,7 +15,7 @@ class Kernel
         console.log 'Kernel spec:', @kernelSpec
         console.log 'Kernel configuration:', @config
         console.log 'Kernel configuration file path:', @configPath
-        @language = @kernelSpec.grammarLanguage
+        @language = KernelManager::getLanguageFor @grammar
```
Comment (n-riesco) [inline lib/kernel.coffee]: We need to add `KernelManager = require './kernel-manager'` to the top of this file.

Comment (n-riesco) [inline lib/kernel.coffee]: This isn't working.

I'm guessing this is caused by the dependency-cycle this statement creates between `KernelManager` and `Kernel`.

Since `Kernel::language` and `Kernel::grammar` aren't used anywhere in the code, I think it's best we remove them and update the code [here](https://github.com/lgeiger/hydrogen/blob/9a9726a677b8b91322457134717bd3b886dd8ee6/lib/kernel.coffee#L14), [here](https://github.com/lgeiger/hydrogen/blob/9a9726a677b8b91322457134717bd3b886dd8ee6/lib/kernel.coffee#L14), [here](https://github.com/nteract/hydrogen/blob/079608feca61a9c419cbcc4445cfc68252a271d7/lib/kernel-manager.coffee#L67) and [here](https://github.com/nteract/hydrogen/blob/079608feca61a9c419cbcc4445cfc68252a271d7/lib/kernel-manager.coffee#L76).

Comment (lgeiger) [inline lib/kernel.coffee]: Oh I'm sorry I didn't restart atom while testing, so this error slipped through.

`Kernel::grammar` is used in this [line](https://github.com/nteract/hydrogen/blob/079608feca61a9c419cbcc4445cfc68252a271d7/lib/watch-view.coffee#L16). I wouldn't remove it.

I think we should switch back to `@language = @kernelSpec.language`
